packages/ingest/recursion.js

```diff
@@ -1,84 +1,93 @@
 'use strict';
 
-const join = require('path').join;
-const log = require('@cumulus/common/log');
-
 /**
- * Handles recursion of a FTP/SFTP list operation
- * It requests a promisified list function that returns contents of
- * a directory on a server
+ * check if segment is a regex segment
  *
- * @param {function} fn The promisified function for listing a directory
- * @param {string} originalPath the full path to use for recursively listing the directory
- * @param {string} [currentPath=null] the current directory to list recursively
- * @param {number} [position=0] current position in the recursive tree
- * @returns {Promise} the promise of an object that has the path is the key and
- *   list of files as values
+ * @param {string} segment - path segment
+ * @returns {boolean}
  */
-async function recursion(fn, originalPath, currentPath = null, position = 0) {
-  // build the recursion path object
-  const regex = /(\(.*?\))/g;
-  const rules = (originalPath || '/').split(regex).map((i) => i.replace(/\\\\/g, '\\'));
-  const map = rules.map((r) => (r.match(regex) !== null));
-
-  let files = [];
-  let path = currentPath;
-  if (!path) {
-    path = rules[position];
-  }
+const isRegexSegment = (segment) => segment.match(/(\([^\)]*\))/g) !== null;
```

Comment:
Can I get an example of what a regex segment looks like? I have trouble following it just by parsing the regex

Reply:
Looks to me like it's looking for any occurrence of `(...)`, where `...` does not contain `\` nor `)`. If that's the case, then perhaps using `match` with the global (`g`) option is unnecessary since a single match would suffice.

To add clarity, perhaps this would help:
```
// Matching open and close parens with no backslash or nested close paren
const NON_NESTED_MATCHING_GROUP = /[(][^\)]*[)]/;

const isRegexSegment = (segment) => NON_NESTED_MATCHING_GROUP.test(segment);
```
or even this point-free version of `isRegexSegment`:
```
const isRegexSegment = NON_NESTED_MATCHING_GROUP.test.bind(NON_NESTED_MATCHING_GROUP);
```
Further, do you also want the prevent having a nested opening paren? If not, then something like `(()` would match, which seems like you wouldn't want.

Reply:
Sorry, I meant to use `CAPTURE`, not `MATCHING`.

Perhaps also consider using the term "captureGroup" instead of "segment" since "capture group" is the correct term for a part of a regex that uses parens to capture parts of a match. The term "segment" is arguably confusing since there is no such term in the regex world regarding the use of parens.

Reply:
This would result in this instead:
```
const NON_NESTED_CAPTURE_GROUP = /[(][^\)]*[)]/;

const isCaptureGroup = (str) => NON_NESTED_CAPTURE_GROUP.test(str);
```

Reply:
Updated. Maintained the use of the word 'segment' because the context is whether this chunk of the path (that was previously split by the matching group) is a regex at all or plaintext, not whether it is specifically a capture group or not.

It's a little confusing because in this use case, all regexes are capture groups by side effect because parentheses is the convention being used to denote a regex section, not because we are needing to capture part of the match.